### PR TITLE
Use HTTP response code 412 for the /health API

### PR DESCRIPTION
**Pull request by calebhailey**
Proposed solution for https://github.com/sensu/sensu/issues/1216 :+1:

### CODE REVIEW

File: lib/sensu/api/process.rb [hunk 1]
```diff
@@ -152,6 +152,10 @@ def not_found!
           ahalt 404
         end
 
+        def precondition_failed!
+          ahalt 412
+        end
+
         def unavailable!
```
Comment (portertech) [inline lib/sensu/api/process.rb]: is unavailable!() used anywhere else? Can it be removed?

**Comment by calebhailey** [comment]
img_20151227_123951

File: lib/sensu/api/process.rb (hunk 1 above)
Comment (calebhailey) [inline lib/sensu/api/process.rb]: @portertech I wasn't sure about that. It looks like it isn't.

**Comment by agoddard** [comment]
\o/

**Comment by calebhailey** [comment]
giphy

**Comment by calebhailey** [comment]
Even though 0.23 has only been out for a few days, this feels like a potentially disruptive change in behavior (e.g. if anyone has `/health` API checks expecting 503 responses for failure), so should we wait to release this until 0.24?

**Comment by portertech** [comment]
@calebhailey API changes must wait for 0.24.

**Comment by calebhailey** [comment]
@portertech cool, makes sense.

**Comment by portertech** [comment]
Will merge, just going to leave open for possible 0.23 bug fixes to go to master.

**Comment by calebhailey** [comment]
Related: https://github.com/sensu/sensu/issues/984

**Comment by portertech** [comment]
@calebhailey could you please close this PR and open a fresh one targeting release/0.24?

**Comment by calebhailey** [comment]
Closing in favor of #1243

**Pull request closed without merging**